### DISCUSSION TITLE
Add the view of comments and make it a first class citizen

### DISCUSSION THREAD

**Discussion by torwag** [astubenbord/paperless-mobile#141] [Feature Requests]
Recently, comments were introduced in paperless-ngx. 
Albeit a nice and an useful feature, I feel it is a bit cumbersome to work with them from a desktop PC. 

One is working on a task, and thinks "ohhh, I need to add this as a comment to a particular document" (e.g. Payed the bill). After that, one has to sit down in front of the PC, open up paperless and adding the comment.

With paperless-mobile, things could become much more smooth and nice. 
If paperless-mobile would not only allow to read and search for comments, but uses the mobile OS functionality to create comments, one can easily and quickly add comments to documents on-the-go. Using the underlying systems for speech-to-text, share, copy-and-paste, etc. E.g. Marking some text in an application and click on "share with..." I can select paperless-mobile, and it will ask me to which document I want to add the text as comment.
 

Furthermore, paperless-mobile could assist in creating comments as seamless as possible, for particular mobile resp. on-the-go situations. Like helping to add time, location, links, source, etc. up to the point that certain functions of paperless-mobile creates automatically (user-adjustable) a comment entry. 

That would allow users to add quickly auxiliary information about a particular document.

**Comment by astubenbord** [comment]
Hi, thanks for the feature proposition.

Comments are something that will be implemented after bulk edits, which I'm currently working on.

> speech-to-text, share, copy-and-paste, etc. E.g. Marking some text in an application and click on "share with..." I can select paperless-mobile, and it will ask me to which document I want to add the text as comment. I can select paperless-mobile, and it will ask me to which document I want to add the text as comment.

Just out of interest, what would the use case be to share text as a comment? Where would these comments come from? But it's an interesting thought, I will think about it, especially since urls linking to images/pdf files could also be shared (in theory), which would create some conflicts.

> Like helping to add time, location, links, source, etc. up to the point that certain functions of paperless-mobile creates automatically (user-adjustable) a comment entry.

Interesting idea, but I think that this kind of feature should be implemented on the server side and paperless-mobile could implement it? Especially since this is highly individual - maybe a model could be trained to help make suggestions on what to add as a comment, just like with correspondents etc.? Or did you mean something different?

**Reply by torwag** [reply]
Hi,
thanks for reading. 
Well, a few examples.

You receive an email with some information about a document you added previously to paperless. Mark this text, select share with and paperless-mobile. As it is only text, paperless-mobile would offer to add it as a comment to a particular document.

You make some notes, during a meeting, on a tablet and want to add some of the information to a document.

You carried out some online banking paying some bills, you want to add the information (receipt of the payment) to the document so, you can easily track bills and payments.

As for the second part. I just thought about some configuration people might set to add to comments.
So instead of adding "Text". Paperless mobile would add "Text, send in from XXXX device, geolocation". Basically, information people would assign with the device, so I think it might be part of paperless-mobile. You could think of some sort of signature added at the end of the actual text, to give more context, why and how that comment was made.
  
In conjunction with paperless-mobile the comment feature could be extended quite a lot. E.g. people might like to add an camera image about a certain status to a comment (like the payment), a link, etc. But indeed this would be something which requires changes on the server side.

Well, all I was thinking, paperless-mobile enables users to use paperless differently. As the main use case was mainly archiving purpose before, users might want to use paperless-mobile to track more what happens to a particular document via paperless-mobile. 
Adding a bill for a device, make notes about missing payment details, make comment about paying, recheck bill for sending in the device under warranty, make notes about tax deductions. Paperless-mobile could be here of great help.

**Reply by astubenbord** [reply]
Thanks for elaborating.

I'm not sure comments are made for some of these use cases (in my opinion at least). For example, references to other documents is a feature that is missing in paperless (I think I've seen you comment on the discussion here https://github.com/paperless-ngx/paperless-ngx/discussions/422) and I think this should be added server side, not on the client side what, I think we agree there.

While comments are essentially free text fields which would allow me to add arbitrary information (e.g. formatted as JSON and parsed in the app separated from your own comments by some unique delimiter), I'm not sure this is the path I want to take since it would sort of force the user to stick to this app in order to properly use and render the information stored in the comments, and I think especially for bulk edits etc. the web client is better suited. But it would indeed allow for some pretty cool features. Maybe they'll add custom fields to documents some day, this would open up a lot of possibilities.


What I could do is implement sharing of text such that once text is received by the app, the user is asked what to do with it (e.g. if a link is shared, the user can choose to upload a document from the given url, or add text as a comment and so on). I'll keep that in mind and think about ways to implement it without causing too much overhead to the overall experience.

Cheers.
